ops/framework.py

```diff
@@ -1010,6 +1045,70 @@ def __get__(self, parent, parent_type=None):
                 self.__class__.__name__, parent_type.__name__))
 
 
+class ConfigHistoryObserver(Object):
+    """A class used to store config history for current application."""
+    _stored = StoredState()
+
+    def __init__(self, parent):
+        super().__init__(parent, None)
+        self._keys = list(self.model.config.keys())
+        self._depth = -1
+        self._stored.set_default(history={key: [self.model.config[key]] for key in self.keys})
+
+    @property
+    def history(self) -> 'StoredDict':
+        """Config history."""
+        return self._stored.history
+
+    @property
+    def keys(self) -> typing.List[str]:
+        """List of tacked keys."""
+        return self._keys
+
+    @keys.setter
+    def keys(self, keys: typing.List[str]) -> None:
+        """Set list if tacked keys."""
+        _keys = []
+        for key in keys:
+            if key not in self.model.config.keys():
+                raise ValueError("Key {} was not found in application config.".format(key))
+
+            _keys.append(key)
+
+        self._keys = keys
+
+    def set_depth(self, depth: int) -> None:
+        """Set depth of tacked history."""
+        if not isinstance(depth, int) and (depth == -1 or 1 <= depth):
+            raise ValueError("Depth must be integer from range [1, inf). [inf <==> -1]")
+
+        self._depth = depth
+
+    def _get_last_value(self, key: str) -> typing.Any:
+        """Get last value for certain key."""
+        key_history = self._stored.history.get(key, [])
+        if not isinstance(key_history, (list, StoredList)) or len(key_history) == 0:
+            raise RuntimeError("Could not get last value for %s", key)
+
+        return key_history[-1]
+
+    def _tack(self, key):
+        """Tack specific key from model config."""
+        value = self.framework.model.config[key]
+        if self._get_last_value(key) != value:
+            logger.debug("tack value %s for %s key", value, key)
+            self._stored.history[key].append(value)
+
+            if self._depth != -1 and len(self._stored.history[key]) > self._depth:
+                # drop old values
+                self._stored.history[key] = self._stored.history[key][-self._depth:]
```

Comment:
`collections.deque`?

Reply:
I never used it before, but it make sense here.
What about changing the depth? `self._stored.history[key] = collenctions.deque(self._stored.history[key], maxlen=depth)`